services/wallet/activity/service.go

```diff
@@ -109,6 +135,40 @@ func (s *Service) GetTxDetails(ctx context.Context, id string) (*EntryDetails, e
 	return getTxDetails(ctx, s.db, id)
 }
 
+func (s *Service) GetLatestBlockNumber(ctx context.Context, chainID uint64) (uint64, error) {
+	s.blkMu.Lock()
+	latestBlockData, ok := s.latestBlockNumbers[chainID]
+	s.blkMu.Unlock()
+	if !ok {
+		return s.fetchLatetBlockNumber(ctx, chainID)
+	}
+	blockDuration, found := w_common.AverageBlockDurationForChain[w_common.ChainID(chainID)]
```

Comment:
If we already fetch it regularly, why not keep and use the history instead of the hardcoded? We can drop the hardcoding and use the fetched all together